frontend/src/metabase/components/EntityItem.jsx

```diff
@@ -52,10 +52,10 @@ const EntityItem = ({
   ].filter(action => action);
 
   return (
-    <EntityItemWrapper py={2} className="hover-parent hover--visibility">
+    <EntityItemWrapper pb={1} pt={2} className="hover-parent hover--visibility">
```

Comment:
Rather than apply specific padding here since `EntityItem` is used a bunch of spots you can tweak the padding where its used. That's the right way to fix the 
> Unfortunately it also does this to the data browse items currently (slightly too much padding above the icon wrapper):

scenario.